modules/social_features/social_event/modules/social_event_max_enroll/src/Controller/UserEnrollInviteControllerAlter.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Drupal\social_event_max_enroll\Controller;
+
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\social_event\Entity\EventEnrollment;
+use Drupal\social_event_invite\Controller\UserEnrollInviteController;
+use Drupal\social_event_max_enroll\Service\EventMaxEnrollService;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Accepts or declines an event enrollment invite.
+ *
+ * @package Drupal\social_event_max_enroll\Controller
+ */
+class UserEnrollInviteControllerAlter extends UserEnrollInviteController {
+
+  /**
+   * The event maximum enroll service.
+   *
+   * @var \Drupal\social_event_max_enroll\Service\EventMaxEnrollService
+   */
+  protected EventMaxEnrollService $eventMaxEnrollService;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    RequestStack $requestStack,
+    AccountProxyInterface $currentUser,
+    EventMaxEnrollService $eventMaxEnrollService
+  ) {
+    parent::__construct($requestStack, $currentUser);
+
+    $this->eventMaxEnrollService = $eventMaxEnrollService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('request_stack'),
+      $container->get('current_user'),
+      $container->get('social_event_max_enroll.service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function updateEnrollmentInvite(EventEnrollment $event_enrollment, string $accept_decline): RedirectResponse {
+    // We should move forward only when the user is accepting the invite.
+    if ($accept_decline === '1') {
+      // Retrieve event ID.
+      $event_id = $event_enrollment->getFieldValue('field_event', 'target_id');
+
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = $this->entityTypeManager()->getStorage('node')->load($event_id);
+
+      if (
+        $node instanceof NodeInterface &&
+        $this->eventMaxEnrollService->isEnabled($node)
+      ) {
+        // Count how many spots are left.
+        $left = $this->eventMaxEnrollService->getEnrollmentsLeft($node);
+
+        // If there are no spots left then we should prevent approving
+        // invite.
+        if ($left === 0) {
```

Comment:
Because you are not using `$left` in any other place, you could simply write it like this:

`if ($event_max_enroll_service->getEnrollmentsLeft($node) === 0)` this is pretty clear

Reply:
Make sense. I have changed it.